File: app/src/main/java/com/robotnec/budget/core/persistence/schema/TransactionSchema.java
package com.robotnec.budget.core.persistence.schema;

import com.yahoo.squidb.annotations.PrimaryKey;
import com.yahoo.squidb.annotations.TableModelSpec;

/**
 * @author zak <zak@swingpulse.com>
 */
@TableModelSpec(
        className = "TransactionRecord",
        tableName = "transactions",
        tableConstraint = "FOREIGN KEY (accountId) REFERENCES account(id)," +
                "FOREIGN KEY (categoryId) REFERENCES category(id)"
)
public class TransactionSchema {

    @PrimaryKey
    private long id;

    private long accountId;
    private long categoryId;
    private String amount;
    private long date;
}

File: app/src/main/java/com/robotnec/budget/core/persistence/schema/TransactionSchema.kt
package com.robotnec.budget.core.persistence.schema

import com.yahoo.squidb.annotations.PrimaryKey
import com.yahoo.squidb.annotations.TableModelSpec

/**
 * @author zak zak@swingpulse.com>
 */
@TableModelSpec(className = "TransactionRecord",
        tableName = "transactions",
        tableConstraint = "FOREIGN KEY (accountId) REFERENCES account(id)," + "FOREIGN KEY (categoryId) REFERENCES category(id)")
class TransactionSchema {

    @PrimaryKey
    private val id: Long = 0
    private val accountId: Long = 0
    private val categoryId: Long = 0
    private val amount: String? = null
    private val date: Long = 0
}
